# Exercise 1

### Step 1. Go to [Kaggle]( https://www.kaggle.com/openfoodfacts/world-food-facts)

### Step 2. Download the dataset to your computer and unzip it.

### Step 3. Use the csv file and assign it to a dataframe called food

In [5]:
from azureml import Workspace

ws = Workspace()
ds = ws.datasets['en.openfoodfacts.org.products.tsv']
food = ds.to_dataframe()

/home/nbuser/anaconda3_23/lib/python3.4/site-packages/IPython/kernel/__main__.py:5: DtypeWarning: Columns (0,3,5,19,20,24,25,26,27,28,36,37,38,39,48) have mixed types. Specify dtype option on import or set low_memory=False.


MemoryError: 

In [ ]:
%matplotlib inline

# pandas bajo el alias pd
import pandas as pd

# numpy bajo el alias np
import numpy as np

# matplotlib.pyplot bajo el alias plt
import matplotlib.pyplot as plt

# seaborn bajo el alias sns
import seaborn as sns

### Step 4. See the first 5 entries

In [ ]:
food.head()

### Step 5. What is the number of observations in the dataset?

In [7]:
food.shape

(356027, 163)

Vemos que hay 356027 observaciones en el data set.

### Step 6. What is the number of columns in the dataset?

Por otro lado, tenemos 163 columnas.

### Step 7. Print the name of all the columns.

In [26]:
food.columns

Index(['code', 'url', 'creator', 'created_t', 'created_datetime',
       'last_modified_t', 'last_modified_datetime', 'product_name',
       'generic_name', 'quantity',
       ...
       'fruits-vegetables-nuts_100g', 'fruits-vegetables-nuts-estimate_100g',
       'collagen-meat-protein-ratio_100g', 'cocoa_100g', 'chlorophyl_100g',
       'carbon-footprint_100g', 'nutrition-score-fr_100g',
       'nutrition-score-uk_100g', 'glycemic-index_100g',
       'water-hardness_100g'],
      dtype='object', length=163)

### Step 8. What is the name of 105th column?

In [25]:
food.columns[149] # Como el índice en python empieza por 0, la culumna 150 tendrá índice 149.

'iodine_100g'

### Step 9. What is the type of the observations of the 105th column?

In [34]:
food.dtypes[149]

dtype('float64')

Vemos que es de tipo float64.

### Step 10. How is the dataset indexed?

In [1]:
food.index

NameError: name 'food' is not defined

### Step 11. What is the product name of the 19th observation?

In [38]:
food['product_name'][18]

'Lotus Organic Brown Jasmine Rice'

Vemos que el nombre del producto de la observación 19 es 'Lotus Organic Brown Jasmine Rice'.